File: GestaoSindicatos/Model/Relatorio.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace GestaoSindicatos.Model
{
    public class Relatorio
    {
        public int Id { get; set; }
        public int NegociacaoId { get; set; }
        [StringLength(100)]
        public string Titulo { get; set; }

        public virtual Negociacao Negociacao { get; set; }
        public virtual ICollection<GrupoPergunta> GruposPerguntas { get; set; }
    }
}

File: front/src/app/model/relatorio.ts
import { Negociacao } from './negociacao';

export enum LayoutGrupo {
    Grupo1Coluna,
    Grupo3Colunas,
    GrupoSemCombo
}

export enum AplicacaoResposta {
    Sim,
    Nao,
    NA
}

export class Relatorio {
    id: number;
    negociacaoId: number;
    titulo: string;

    negociacao: Negociacao;
    gruposPerguntas: GrupoPergunta[];
}

export class GrupoPergunta {
    id: number;
    relatorioId: number;
    ordem: number;
    texto: string;
    layoutGrupo: LayoutGrupo;

    respostas: RespostaRelatorio[];
}

export class RespostaRelatorio {
    id: number;
    ordem: number;
    pergunta: string;
    resposta: string;
    aplicacaoResposta: AplicacaoResposta;
    grupoPerguntaId: number;
    numColunas: number;

    grupoPergunta: GrupoPergunta;

}
